tracer/src/Datadog.Trace/Telemetry/TelemetryTransportManagerV2.cs

```diff
@@ -5,105 +5,129 @@
+
+        var agentProxy = _transports.AgentTransport;
+        var agentless = _transports.AgentlessTransport;
+
+        // If only one transport is configured, continue to use it
+        // If we're using agentProxy, and agentless is configured, use that
+        // If we're using agentless, and agentProxy is configured, and we don't _know_ that we can't use it, use that
+        // If no transports are available,
+        if (currentTransport == agentProxy)
         {
-            _initialFatalCount++;
-            if (_initialFatalCount >= MaxFatalErrors)
-            {
-                // we've had MaxFatalErrors 404s, 1 minute apart, prob never going to work, so try next transport
-                return ConfigureNextTransport();
-            }
+            return agentless is null
+                       ? agentProxy // nothing else available, keep using it
+                       : agentless;  // switch from agent to agentless
         }
-        else
+
+        Debug.Assert(agentless is not null, "If current transport is not agent, it must be agentless");
+
+        if (agentProxy is null)
         {
-            _failureCount++;
-            if (_failureCount >= MaxTransientErrors)
-            {
-                // we've had MaxTransientErrors in a row, 1 minute apart, probably something bigger wrong
-                return ConfigureNextTransport();
-            }
+            // nothing else available, keep using it
+            return agentless!;
         }
 
-        // We should retry next time
-        return TelemetryTransportResult.TransientError;
+        return _canSendToAgent != false
+                   ? agentProxy // might be able to send to agent, so try it
+                   : agentless!; // the agent is not available, so stick to agentless
     }
 
-    private TelemetryTransportResult ConfigureNextTransport()
+    private void HandleAgentDiscoveryUpdate(AgentConfiguration config)
     {
-        // if we have more transports available, treat this as a transient error
-        // otherwise it's fatal
-        _currentTransport++;
-
-        if (_currentTransport < _transports.Length)
+        _canSendToAgent = !string.IsNullOrWhiteSpace(config.TelemetryProxyEndpoint);
+        if (_canSendToAgent == true)
         {
-            Log.Debug(
-                "Telemetry transport failed. Using next transport {TransportInfo}",
-                _transports[_currentTransport].GetTransportInfo());
-
-            // reset the circuit breaker counters
-            _failureCount = 0;
-            _initialFatalCount = 0;
-            _hasSentSuccessfully = false;
-            // try with the next transport next time
-            return TelemetryTransportResult.TransientError;
+            Log.Debug("Telemetry agent proxy is available.");
+        }
+        else
+        {
+            Log.Warning("Detected agent does not support telemetry agent proxy");
```

Comment:
I'd say it's not a warning. 
```suggestion
            Log.Information("Detected agent does not support telemetry agent proxy");
```

Reply:
👍  Yeah, tbh, I'm even inclined to say debug 🤔  I think this is mostly copy-pasta from the stats aggregate implementation I was following, but it makes sense to be a warning in that case, because you try to explicitly enable it, but you can't